src/main/java/at/hannibal2/skyhanni/features/event/spook/TheGreatSpook.kt

```diff
@@ -53,11 +62,80 @@ object TheGreatSpook {
+    )
+
+    private fun solveMath(math: String): String {
+        val mathFormatted = math.replace("x", "*").replace(" ", "")
+        return try {
+            val result = MathUtils.evaluateExpression(mathFormatted)
+            result.toString()
+        } catch (e: Exception) {
+            "§c§lError"
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isSolverEnabled()) return
+
+        val message = event.message
+
+        if (isMathSolverEnabled()) {
+            mathFearMessagePattern.matchMatcher(message) {
+                val math = group("math")
+                val result = solveMath(math)
+                DelayedRun.runNextTick {
+                    val feature: KMutableProperty0<*> = config.primalFearSolvers::mathPrimalFear
+                    ChatUtils.clickToActionOrDisable(
+                        "The result is: $result",
+                        feature,
+                        actionName = "Send the result",
+                        action =
+                        {
+                            HypixelCommands.allChat(result)
+                        },
+                    )
+                }
+            }
+        }
+
+        if (isSpeakingSolverEnabled()) {
+            speakingFearMessagePattern.matchMatcher(message) {
+                DelayedRun.runNextTick {
+                    val feature: KMutableProperty0<*> = config.primalFearSolvers::publicSpeakingPrimalFear
+                    ChatUtils.clickToActionOrDisable(
+                        "Click to send a random string to complete the Primal Fear",
+                        feature,
+                        "Sends the message!",
+                        action =
+                        {
+                            HypixelCommands.allChat("I looove SkyHanni!" + StringUtils.generateRandomString(5))
+                        }
+                    )
+                }
+            }
+        }
+    }
+
     private fun isTimerEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearTimer
 
     private fun isNotificationEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearNotification
     private fun isFearStatEnabled(): Boolean = LorenzUtils.inSkyBlock && config.fearStatDisplay
     private fun isTimeLeftEnabled(): Boolean = LorenzUtils.inSkyBlock && config.greatSpookTimeLeft
 
-    private fun isAllDisabled(): Boolean = !isTimeLeftEnabled() && !isTimerEnabled() && !isFearStatEnabled() && !isNotificationEnabled()
+    private fun isSolverEnabled(): Boolean = isMathSolverEnabled() || isSpeakingSolverEnabled()
+
+    private fun isMathSolverEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearSolvers.mathPrimalFear
+    private fun isSpeakingSolverEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearSolvers.publicSpeakingPrimalFear
+
+    private fun isAllDisabled(): Boolean = !isTimeLeftEnabled() && !isTimerEnabled() && !isFearStatEnabled() &&
```

Comment:
very confusing

Reply:
To add onyo this, the existing code before you made this feature was also bad but adding another 3 options was the cherry on top. Maybe look at how many other features have a global enabled value and implemetn that instead

Reply:
Removed entirely as it isnt needed